src/general/Mutations.cs

```diff
@@ -305,6 +313,10 @@ private void MutateBehaviour(MicrobeSpecies parent, MicrobeSpecies mutated)
             if (organelles.Count < 1)
                 organelles.Add((OrganelleTemplate)parentOrganelles[0].Clone());
         }
+
+        // Try again if the mutation has islands
+        if (organelles.GetIslandHexes().Any())
```

Comment:
This still isn't in the place where I would put this. I'd put this to the add organelle and remove organelle methods.

Reply:
But if I run the code per modification it gets executed more than if I validate once in the end. Wouldn't this be less performant?

Reply:
I think it is better to fail early so that the already included place retry logic can try again.
It is a bit less performant if you use a general "are there islands" code, however tweaking the island check to only check the newly placed organelle can be done.

Reply:
One more thing I remembered, before going to sleep, is that the add organelle code *only* adds organelles adjacent to existing ones, so if you verify that's the case, the island check doesn't need to be there. I suggest the island check is put in
- the remove organelle mutation place (with a few retries to make it more likely to find a valid organelle to remove)
- at the end, but just to log an error (this is to guard against changes someone might make to the add organelle mutation code in the future)

Reply:
the existing code only allowes new organelles to be placed next to an existing one.
The only place where I have to check for islands is deletion.
For deletion, I have not found any way to optimize the CheckIslands method.
Therefore, calling the method once is more practical and more performant than calling it many times.

Reply:
Due to the code copying one organelle (or does not if an organelle gets removed) at a time into a new list it is pretty difficult to check for islands in runtime, because I would have to
- add the organelles to the originalOranelles
- remove the organelle from originalOrganelles
- check for islands
- restore originalOrganelles

this means I would have to copy the originalOrganelles for every removal in every mutation, which sounds very ineffective.

Reply:
Can't you just do logic like:
- for i in some number of attempts, maybe 3-4 is good
- remove organelle
- check for islands
- if there are islands put the organelle back
- if there aren't islands, break the loop

Reply:
not really, because the code does not create a copy to keep the original and modify the copy, but it clears the list, iterates through the original list and creates or removes organelles at random.
After the random event, the code copies the original entry into the cleared list if it wasn't removed. Therefore the new list slowly gets built, which is not compatible with the checkislands code

Reply:
Then change the code to copy all the organelles first, then iterate over them and randomly try removing a few of them.

Reply:
see this line:
https://github.com/Revolutionary-Games/Thrive/blob/45a74ee0008aa2f5d6888ed6b10ff4bb007524a1/src/general/Mutations.cs#L260

Reply:
Okay fine, I give up. Do the approach where the entire algorithm is just reattempted if there are islands at the end but put like a limit of 10 there to not infinitely get stuck in that loop (and log that as a failure). If that results in the mutation code not find valid mutations often enough the limit might need to be increased **or** the algorithm actually needs to be rewritten as I've explained to properly be able to generate mutations while checking for islands.